Classes/Domain/Model/Changes/AbstractMove.php

```diff
@@ -67,4 +67,20 @@ protected function generateUniqueNodeName(NodeInterface $parentNode)
         return $this->contentRepositoryNodeService
             ->generateUniqueNodeName($parentNode->getPath());
     }
+
+    /**
+     * Generate a unique node name for the copied node if needed
+     *
+     * @param NodeInterface $parentNode
+     * @param NodeInterface $node
+     * @return string
+     */
+    protected function generateUniqueNodeNameIfNeeded(NodeInterface $parentNode, NodeInterface $node)
+    {
+        if ($this->contentRepositoryNodeService->nodePathAvailableForNode($parentNode->getPath() . $node->getName(), $node)) {
+            return $node->getName();
+        }
```

Comment:
Should'nt this simple return 'null' if no new name is needed.

Reply:
Other than that the PR works for the cases i see.

Reply:
`null` ends with the same result but is better more transparent :+1: